### ISSUE TITLE
Allow ships to use hyperdrives instead of jump drives when they are more efficient fuel-wise

### ISSUE THREAD

**Issue by Bladewood** [endless-sky/endless-sky#1214]
And by that, I mean when they use less fuel. Hyperspace paths are rarely better than jump drive paths, but when they are (such as with Ruchbah-Sheratan), they should be used.

On a related note, if this is implemented, perhaps you should integrate a hyperdrive into the jump drive.

**Comment by MarcelineVQ** [comment]
Outfitting random ships in my fleet of jumpdrivers in Ruchbach with hyperdives as well has lead me to conclude that:
Hyperdrive will be used over jump drive if both would use the same nodes because it will use less fuel.

I'm reasonably sure jump drive has a hyperdrive built in as well because I can make the Ruchbah-Sheratan jump in one jump and it's clearly out of range otherwise. It will use 200 fuel though whereas a hyperdrive would have used 100, which I personally don't mind at all.
8 mass is less than 20 after all if I had to compare a fuel tank to a hyperdrive.

**Comment by Amazinite** [comment]
I'm pretty sure that the jump pathing already prefers hyperdrive over jumping if it's more fuel efficient. According to MZ in #1160 

> The route-finding is based on amount of fuel needed

What kind of path causes it to auto plan jumping over hyperdrive when hyperdrive is more efficient?

**Comment by Bladewood** [comment]
> I'm reasonably sure jump drive has a hyperdrive built in as well because I can make the Ruchbah-Sheratan jump in one jump with just a jumpdrive and it's clearly out of range otherwise.

>It will use 200 fuel though whereas a hyperdrive would have used 100

Yes, those are what I am concerned about. A hyperdrive must be necessary for that route to be jumped without taking the nearby systems to it, and if it takes the same fuel as a jump drive, that's a problem for me, because even the smallest jump uses up precious fuel that could be used for, I dunno, a cloaking device when you lack a decent ramscoop. Specifically, lots of wandererscoops (yay, I have coined a new term that will probably not be used by anyone but me). On that note, this is somewhat important for the Wanderer worlds because in the beginning, you are not allowed to land and refuel except at their homeworld. If we integrate a hyperdrive, we use less fuel, so more jumps can be made, and we can actually get to their homeworld without the kindness of your alien hosts or a decent ramscoop.

**Comment by Amazinite** [comment]
I don't see the issue here then. You're using a Jump Drive to allow you to jump to places you normally couldn't or get from point A to point B faster for twice the fuel. That's the trade off of having a Jump Drive. You could easily install both a Hyperdrive and Jump Drive to avoid the problem.

**Comment by Bladewood** [comment]
...wait, nevermind.
I did that a long time ago, but it's different now.
Apparently my game was trying to confuse me the last time I did it, because at that time, dual wielding drives seemed to use jump drives anyway, but I just checked, and it used my hyperdrive this time.
